scipy/sparse/linalg/_special_sparse_arrays.py

```diff
@@ -0,0 +1,394 @@
+    on a uniform rectangular grid in `N` dimensions given by tuple `shape`
+    and output its eigenvalues.
+
+    The Laplacian matrix `L` is square real symmetric negative definite.
+
+    Parameters
+    ----------
+    shape : tuple of int
+        A tuple of length `N` (corresponding to the dimension of the Lapacian),
+        where each entry gives the size of that dimension. The Laplacian matrix
+        is square of the size np.prod(shape)
+    boundary_conditions: ``'Dirichlet'`` or ``'Neumann'`` or ``'Periodic'``
+        The type of the boundary conditions on the boundaries of the grid
+    dtype: one of numeric numpy dtypes
+        The dtype of the matrix output.
+        Default: ``np.int32``.
+
+    Returns
+    -------
+    lap_Nd: `LinearOperator` custom object
+        The object containing the output standard for `LinearOperator`: e.g.,
+        lap_Nd.toarray() : (n, n) ndarray
+        lap_Nd.tosparse() : (n, n) DIAgonal sparse format
+        lap_Nd._matvec and lap_Nd._matmat: callable objects
+            The handles to functions that multiply the Laplacian matrix `L` of
+            the shape `n`-by-`n` on the right by an input matrix `x` of
+            the shape `n`-by-`k` to output ``L @ x`` without constructing `L`
+    lap_Nd.eigenvalues : (n, ) ndarray, float
+        Eigenvalues of the Laplacian matrix ordered ascending
+
+    laplacian_obj: class
+        The object containing the output
+    laplacian_obj.sparse : csr_array
+        The ``(n, n)`` Laplacian matrix in a Compressed Sparse Row (CSR) format
+    laplacian_obj : callable
+        The handle to a function that multiplies the Laplacian matrix
+        `L` of the shape `n`-by-`n` on the right by an input matrix `x`
+        of the shape `n`-by-`k` to output ``L @ x`` without constructing `L`
+    laplacian_obj.eigenvalues : (n, ) ndarray, float
+        Eigenvalues of the laplacian matrix (pair) ordered ascending
+
+    .. versionadded:: 1.12.0
+
+    Notes
+    -----
+    Compared to the MATLAB/Octave implementation [1] of 1-, 2-, and 3-D
+    Laplacian, this code allows the arbitrary N-D case and the matrix-free
+    callable option, but is currently limited to pure Dirichlet, Neumann or
+    periodic boundary conditions only and outputs just the eigenvalues,
+    no eigenvectors.
+
+    The Laplacian matrix of a graph (`scipy.sparse.csgraph.laplacian`) of a
+    rectangular grid corresponds to the negative Laplacian with the Neumann
+    conditions, i.e., ``boundary_conditions = 'Neumann'``.
+
+    All eigenvalues of the discrete Laplacian operator for
+    an ``N``-dimensional  regular grid of shape `shape` with ``h=1`` are
+    analytically known [2].
+
+    References
+    ----------
+    .. [1] https://github.com/lobpcg/blopex/blob/master/blopex_\
+tools/matlab/laplacian/laplacian.m
+    .. [2] https://w.wiki/7CVT
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.sparse.linalg import Laplacian_Nd
```

Comment:
It appears there's still some inconsistencies w.r.t. `Laplacian_Nd` vs. `LaplacianNd`.

Reply:
should finally be fixed now